core/src/main/scala/org/apache/spark/util/RpcUtils.scala

```diff
@@ -47,14 +46,24 @@ object RpcUtils {
   }
 
   /** Returns the default Spark timeout to use for RPC ask operations. */
-  def askTimeout(conf: SparkConf): FiniteDuration = {
-    conf.getTimeAsSeconds("spark.rpc.askTimeout",
-      conf.get("spark.network.timeout", "120s")) seconds
+  def askTimeout(conf: SparkConf): RpcTimeout = {
+    try {
+      RpcTimeout(conf, "spark.rpc.askTimeout")
+    }
+    catch {
+      case _: Throwable =>
+        RpcTimeout(conf, "spark.network.timeout", "120s")
```

Comment:
Can you change this to catch a much narrower exception?

Or even better -- how about you move this into `RpcTimeout.apply` -- have it take a prioritized list of properties to check, and it will check the conf for each of them, avoiding any need for exception handling